### ISSUE TITLE
Error when mapping with injected service

### ISSUE THREAD

**Issue by jornhd** [agileobjects/AgileMapper#204]
AgileMapper Version 1.7.1
I'm having problems when trying to use an injected service in member mapping.
Using .NET core 3.1, I've setup the service provider in public static void Configure(IApplicationBuilder app):
```
            Mapper.WhenMapping
                .UseServiceProvider(app.ApplicationServices);
```

This seems to work fine as I get a logger message in mapping configuration:
```
    public class MyDtoMapper : MapperConfiguration
    {
        protected override void Configure()
        {
            GetService<ILogger<MyDtoMapper>>().LogInformation("Hello, from MyDtoMapper");
            WhenMapping....
        }
    }
```

I've tried this from the documentation:

> **A value from an injected service:**
> 
> ```
> // Retrieve an IDateTimeProvider instance from a configured
> // service provider, and use its UtcNow value:
> Mapper.WhenMapping
>     .From<OrderDto>()        // Apply to OrderDto mappings
>     .ToANew<Order>()         // Apply to Order creation
>     .Map(ctx => ctx.GetService<IDateTimeProvider>().UtcNow)
>     .To(o => o.DateCreated); // o is the Order
> ```

But this results in the following exception.

_System.InvalidOperationException:
variable 'i' of type 'System.Int32' referenced from scope '', but it is not defined
   at System.Linq.Expressions.Compiler.VariableBinder.Reference(ParameterExpression node, VariableStorageKind storage)_

So.. I'm I doing something wrong, or is this a bug?

**Comment by SteveWilkes** [comment]
That sounds like a bug - I'll check it out. Thanks for letting me know!

Steve

**Comment by SteveWilkes** [comment]
Hi!

I've had a go and been unable to reproduce this - [here's](https://dotnetfiddle.net/QaueiY) a fiddle which successfully maps using `GetService<>`.

Could you share exactly what you're trying to map, please?

All the best,

Steve

**Comment by jornhd** [comment]
Modified your fiddle, and now get the same error: https://dotnetfiddle.net/4XzT0u

**Comment by SteveWilkes** [comment]
This is fixed in release 1.7.2, which is now [on NuGet](https://www.nuget.org/packages/AgileObjects.AgileMapper).

Thanks again!
